### ISSUE TITLE
smart_ plugin

### ISSUE THREAD

**Issue by svirusxxx** [munin-monitoring/munin#602]
On gentoo systems witch kernels > 4.0 I have problem witch smart_ plugin.

```
$ munin-run --debug smart_sda
# Processing plugin configuration from /etc/munin/plugin-conf.d/munin-node
# Setting /rgid/ruid/ to /120/0/
# Setting /egid/euid/ to /120 120/0/
# Setting up environment
# About to run '/etc/munin/plugins/smart_sda'
smart_sda: plugins' UID: 0 / plugins' GID: 120
smart_sda: No S.M.A.R.T device name found in plugin's symlink!
```

but /dev/sda exist , and smartcl works perfect.
also I use ssd_life plugin that using smartclt, and don't have any problems.

**Comment by dipohl** [comment]
Please increase verbosity by changing False to True in the plugins source:

  ## You may edit the following 3 variables
  # Increase verbosity (True/False)
  verbose=False

and tell resulting output of:

  # munin-run --debug smart_sda config
  # munin-run --debug smart_sda

What version of munin-node are you using?

**Comment by svirusxxx** [comment]
it's output witch verbose=True
```
# munin-run --debug smart_sda config
# Processing plugin configuration from /etc/munin/plugin-conf.d/munin-node
# Setting /rgid/ruid/ to /244/0/
# Setting /egid/euid/ to /244 244 6/0/
# Setting up environment
# About to run '/etc/munin/plugins/smart_sda config'
smart_sda: plugins' UID: 0 / plugins' GID: 244
smart_sda: No S.M.A.R.T device name found in plugin's symlink!
```
```
# munin-run --debug smart_sda
# Processing plugin configuration from /etc/munin/plugin-conf.d/munin-node
# Setting /rgid/ruid/ to /244/0/
# Setting /egid/euid/ to /244 244 6/0/
# Setting up environment
# About to run '/etc/munin/plugins/smart_sda'
smart_sda: plugins' UID: 0 / plugins' GID: 244
smart_sda: No S.M.A.R.T device name found in plugin's symlink!
```

munin-2.0.19 (stable in gentoo)
OS: Gentoo
Kernel: 4.1.7

**Comment by dipohl** [comment]
Is there something like SELinux enabled on your server?

**Comment by svirusxxx** [comment]
No, this is standard kernel compiled myself + patches from gentoo

On 5 December 2015 at 21:54, dipohl <notifications@github.com> wrote:

> Is there something like SELinux enabled on your server?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/munin-monitoring/munin/issues/602#issuecomment-162244750>
> .
>



-- 
Piotr (ViruS) Sikora
E-mail/JID: svirus@gmail.com
http://virus.hostv.pl/

**Comment by dipohl** [comment]
what shows 
  # ls -l /etc/munin/plugins/smart_sda

and 
  # ls -l /dev/sda

and please add the following additional print command in the plugins code
and tell the result with verbose = True

  def get_hard_drive_name() :
      global plugin_name
      try :
          name=[plugin_name[string.index(plugin_name,'_')+1:]]
          verboselog('Device Name: '+name[0])

**Comment by ssm** [comment]
I'm unable to reproduce this bug.

Tested with Linux 4.3, smartmontools 6.4+svn4214-1, and the munin plugin from munin 2.x (which is the same in devel)

**Comment by svirusxxx** [comment]
I upgraded all my servers to kernel  4.3.3 (stable in gentoo) and it works for me now.

**Comment by delacroix0815** [comment]
I am running gentoo with gentoo-kernel 4.4.6 and i can't confirm that it's running. I still get smart_sda: No S.M.A.R.T device name found in plugin's symlink! when running munin-run --debug --pidebug smart_sda. My munin version is 2.0.19-r1. Here's the complete output:

# Processing plugin configuration from /etc/munin/plugin-conf.d/munin-node
# Setting /rgid/ruid/ to /996/0/
# Setting /egid/euid/ to /996 996 6/0/
# Setting up environment
# Environment smartargs = -H -c -l error -l selftest -l selective
# About to run '/etc/munin/plugins/smart_sda config'
smart_sda: plugins' UID: 0 / plugins' GID: 996
smart_sda: No S.M.A.R.T device name found in plugin's symlink!

**Comment by dipohl** [comment]
What shows 
  # ls -l /etc/munin/plugins/smart_sda

and 
  # ls -l /dev/sda

and please add the following additional print command in the plugins code
and tell the result with verbose = True

  def get_hard_drive_name() :
      global plugin_name
      try :
          name=[plugin_name[string.index(plugin_name,'_')+1:]]
          verboselog('Device Name: '+name[0])

**Comment by delacroix0815** [comment]
Output of ls -l /etc/munin/plugins/smart_sda:
lrwxrwxrwx 1 root root 33  6. Apr 23:56 /etc/munin/plugins/smart_sda -> /usr/libexec/munin/plugins/smart_

Output of ls -l /dev/sda:
brw-rw---- 1 root disk 8, 0  5. Jul 23:17 /dev/sda

I have added the verboselog('Device Name: '+name[0]) line and changed Verbose=true in /usr/libexec/munin/plugins/smart_ , this is the output from munin-run smart_sda --pidebug --debug :

# Processing plugin configuration from /etc/munin/plugin-conf.d/munin-node
# Setting /rgid/ruid/ to /996/0/
# Setting /egid/euid/ to /996 996 6/0/
# Setting up environment
# About to run '/etc/munin/plugins/smart_sda'
smart_sda: plugins' UID: 0 / plugins' GID: 996
smart_sda: No S.M.A.R.T device name found in plugin's symlink!

It looks like the new verboselog line doesn't get called at all ?

**Comment by velaar** [comment]
Have exactly the same bug with exactly the same outputs. By removing the exception catch was able to get this error: 

```
Traceback (most recent call last):
  File "/etc/munin/plugins/smart_sda", line 403, in <module>
    hard_drive=get_hard_drive_name()
  File "/etc/munin/plugins/smart_sda", line 232, in get_hard_drive_name
    name=[plugin_name[string.index(plugin_name,'_')+1:]]
AttributeError: 'module' object has no attribute 'index'
```

Not sure if it is in any way relevant.

**Comment by dipohl** [comment]
Velaar,

thanks for your investigation! 

I sent an email with your findings to the original plugin author and hoped, that he will make the needed changes to enable the code to work with the new python version. But unfortunately his mail address is no longer existent.

So we have to find another contributor for this plugin as it seems.

fyi and cheers,

Gabriele

**Comment by sumpfralle** [comment]
This issue is fixed now with #728 and 1fda8fb3e58e7fa6333eb8c4fd888e75fa182ed9 in `master` as well as in `stable-2.0`.
